### PR TITLE
TritonTemplate dtype fixes

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141991

- Set the dtype of "acc" appropriately so that epilogue fusion will have args with dtype
- Update dtype propagation to use `type_to_dtype` instead of instantiating tensor
- Throw if we have a string arg where we should have a proper CSEVariable, unless we're doing the Modification Subgraph thing which is nyi. everything else is appropriately typed (cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @drisspg ).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/common.py [hunk 1]
```diff
@@ -1561,7 +1561,7 @@ def namedvar(
         bounds: ValueRanges[Any] = ValueRanges.unknown(),
         dtype: Optional[torch.dtype] = None,
     ) -> CSEVariable:
-        assert name not in self.varname_map, "duplicate name"
+        torch._check(name not in self.varname_map, lambda: f"duplicate name: {name}")
```
Comment (Skylion007) [inline torch/_inductor/codegen/common.py:1564]:
```suggestion        
        torch._check_value(name not in self.varname_map, lambda: f"duplicate name: {name}")
```
So it raises a ValueError

File: torch/_inductor/codegen/triton.py [hunk 2]
```diff
@@ -657,22 +657,6 @@ def _print_RoundDecimal(self, expr):
 
 texpr = TritonPrinter().doprint
 
-# correct cases where Triton types names don't match PyTorch
```
Comment (drisspg) [inline torch/_inductor/codegen/triton.py:660]: love it

**Review by drisspg — APPROVED** [review]

**Comment by blaine-rister** [comment]
LGTM!

**Pull request closed without merging**